### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by mhamilton723**

### CODE REVIEW

**Comment by mhamilton723** [comment]
@imatiach-msft when you get the chance could you make a doc like this for lightgbm? I can also help if you send me the information and rough schematics

File: docs/http.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+# HTTP on Spark
+### A library for interacting with HTTP services from Spark
+
+- **Flexible**: Encodes the entire HTTP protocol in spark for full control of web requests
```
Comment (drdarshan) [inline docs/http.md]: s/spark/Apache Spark/

File: docs/http.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+- **Performant**: Fully distributed across workers, built in support for multi-threaded buffering, batching, and Asynchronous request concurrency.
+- **Easy**: High level APIs for automatic parsing of requests, abstracting all HTTP knowledge to under the hood. 
+- **Acessible**: Usable in Python and Scala. Native integration with Scala's [Apache HTTP Core](https://hc.apache.org/httpcomponents-core-ga/httpcore/apidocs/org/apache/http/package-summary.html). Native integration with [Python Requests](http://docs.python-requests.org/en/master/) library coming soon!
```
Comment (drdarshan) [inline docs/http.md]: Accessible from multiple languages.

File: docs/http.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
+# HTTP on Spark
+### A library for interacting with HTTP services from Spark
+
+- **Flexible**: Encodes the entire HTTP protocol in spark for full control of web requests
+- **Performant**: Fully distributed across workers, built in support for multi-threaded buffering, batching, and Asynchronous request concurrency.
```
Comment (drdarshan) [inline docs/http.md]: async instead of Async.

File: docs/http.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+- **Easy**: High level APIs for automatic parsing of requests, abstracting all HTTP knowledge to under the hood.
```
Comment (drdarshan) [inline docs/http.md]: Easy to use.

Comment (drdarshan) [reply docs/http.md]: High-level

File: docs/http.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+client = SimpleHTTPTransformer()\
+  .setInputCol("data")\
+  .setOutputParser(JSONOutputParser().setDataType(StructType().add("replies", StringType)))\
+  .setUrl("www.my_service_url.com/any_api_here")\
+  .setOutputCol("results")
+
+responses = client.transform(df)
+```
+## High Performance Functionality
+The Simple HTTP transformer provides options to easily opt in to dynamic batchign of requests and asynchronous sending simply use the corresponding settings. By default these options are not enabled for simplicity and easier debugging.
```
Comment (drdarshan) [inline docs/http.md]: batching

File: docs/http.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+  .setOutputParser(JSONOutputParser().setDataType(StructType().add("replies", StringType)))\
+  .setUrl("www.my_service_url.com/any_api_here")\
+  .setOutputCol("results")
+
+responses = client.transform(df)
+```
+## High Performance Functionality
+The Simple HTTP transformer provides options to easily opt in to dynamic batchign of requests and asynchronous sending simply use the corresponding settings. By default these options are not enabled for simplicity and easier debugging. 
+
+- MaxBatchSize: Parameter that enables buffered minibatching. If this parameter is set, a background thread will fetch up to at most `MaxBatchSize` requests and batch them into a single request by creating an Array of their entity data. The method can send less that maxBatchSize requests if one tries to matrialize the iterator quickly or there is not that much data in each partition. The behavior that results from this is that each new request sends all of the new data that has accumulated at this stage of the pipeline.
```
Comment (drdarshan) [inline docs/http.md]: `MaxBatchSize` instead of MaxBatchSize?

File: docs/http.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+- Concurrency: This parameter allows one to send up to `Concurrency` requests simultaneously using scala futures under the hood. If this parameter is set to 1 (default) then no scala futures are used
```
Comment (drdarshan) [inline docs/http.md]: Scala instead of scala

File: docs/mmlspark-serving.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
+# <img src="https://mmlspark.blob.core.windows.net/graphics/SparkServing3.svg?sanitize=true" width="90"> MMLSpark Serving
+### An engine for deploying Spark jobs as distributed web services 
+
+
+- **Distributed**: Takes full advantage of Node, JVM, and thread level parallelism that Spark is famous for
+- **Fast**: No single node bottlenecks, no round trips to python, and requests can be routed directly to and from worker JVMs through network switches. Spin up a web service in a matter of seconds
```
Comment (drdarshan) [inline docs/mmlspark-serving.md]: Python

Comment (drdarshan) [reply docs/mmlspark-serving.md]: matter of seconds.

File: docs/mmlspark-serving.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+    .option("name", "my_api")\    
+    .queryName("my_query")\       
+    .option("replyCol", network.getOutputCol())\ 
+    .option("checkpointLocation", "file:///path/to/checkpoints")\
+    .start()
+```
+
+## Architecture
+
+MMLSpark serving adds  spectial streaming sources (and paired streaming sinks) to turn any streaming job into a web service. MMLSpark serving comes with two deployment options that vary based on the load balancing used.
```
Comment (drdarshan) [inline docs/mmlspark-serving.md]: special

File: docs/http.md (hunk 1 above)
Comment (mhamilton723) [reply docs/http.md]: Sorry I dont quite understand what you mean here

Comment (drdarshan) [reply docs/http.md]: Sorry, replace spark with Apache Spark: https://en.wikipedia.org/wiki/Sed#Substitution_command

File: docs/http.md (hunk 2 above)
Comment (mhamilton723) [reply docs/http.md]: Done

File: docs/http.md (hunk 1 above)
Comment (mhamilton723) [reply docs/http.md]: Done

File: docs/http.md (hunk 3 above)
Comment (mhamilton723) [reply docs/http.md]: Done

File: docs/http.md (hunk 4 above)
Comment (mhamilton723) [reply docs/http.md]: Done

File: docs/http.md (hunk 5 above)
Comment (mhamilton723) [reply docs/http.md]: Done

File: docs/http.md (hunk 6 above)
Comment (mhamilton723) [reply docs/http.md]: Done

File: docs/http.md (hunk 7 above)
Comment (mhamilton723) [reply docs/http.md]: Done

File: docs/mmlspark-serving.md (hunk 8 above)
Comment (mhamilton723) [reply docs/mmlspark-serving.md]: Done

File: docs/mmlspark-serving.md (hunk 9 above)
Comment (mhamilton723) [reply docs/mmlspark-serving.md]: Done

**Comment by mhamilton723** [comment]
Checked with Acrolinx and both docs have >=90

**Pull request merged**